app/models/miq_server.rb

```diff
@@ -346,6 +351,15 @@ def monitor
     end
   end
 
+  def monitor_myself
+    if memory_usage.to_i > memory_threshold
+      msg = "server(pid: #{pid}, name: #{name}) memory usage [#{memory_usage.to_i}] exceeded limit: [#{memory_threshold}].  Exiting server process."
+      _log.warn(msg)
+      MiqEvent.raise_evm_event_queue_in_region(self, "evm_server_memory_exceeded", :event_details => msg, :type => self.class.name)
```

Comment:
I think a notification might be warranted for this, thoughts?

I find them to be a bit more user friendly than `MiqEvents`